docs/concepts/task-graphs.rst

```diff
@@ -118,33 +118,111 @@ Dependencies
 ------------
 
 Dependencies between tasks are represented as labeled edges in the task graph.
-For example, a test task must depend on the build task creating the artifact it
-tests, and this dependency edge is named 'build'. The task graph generation
-process later resolves these dependencies to specific task ids.
+They are specified via the ``dependencies`` key which is an object of the form
+``{ "<edge>": "<label>"}``. The ``edge`` is an arbitrary name which can be used
+to refer to the dependency later on. The ``label`` is the task label of the
+dependency.
```

Comment:
I may be misremembering, but the value can also be a Task object, no?

Reply:
I'm not sure.. The `task_description_schema` allows it, but I'm failing to find where we translate from `Task` object -> label (as the latter is definitely what's expected). Maybe it used to be supported but got removed at some point?